telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatch.java

```diff
@@ -23,4 +24,53 @@ public MetricBatch(Collection<Metric> metrics, Attributes commonAttributes) {
   public TelemetryBatch<Metric> createSubBatch(Collection<Metric> telemetry) {
     return new MetricBatch(telemetry, getCommonAttributes());
   }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private String instrumentationProvider;
+    private String serviceName;
+    private Collection<Metric> metrics = Collections.emptyList();
+    private CommonAttributesBuilder commonAttributesBuilder = new CommonAttributesBuilder();
+
+    /**
+     * Optional. Specify the name of the service that is creating the metrics. The service name will
+     * be included in all common attributes.
+     *
+     * @param serviceName - The name of the service
+     */
+    public Builder serviceName(String serviceName) {
+      commonAttributesBuilder.serviceName(serviceName);
+      return this;
+    }
+
+    /**
+     * Optional. Specify the name of the instrumentation that provides the metrics. The
```

Comment:
I think examples will be useful in these docs.  Also, it's actually not the name of the instrumentation. It's the library that provides the instrumentation APIs.  For example, "micrometer" would be the instrumentation provider, which is a general instrumentation library, not a provider of specific instrumentation.

Reply:
Agreed.